std/typecons.d

```diff
@@ -149,25 +157,37 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
-        _p = null;
+        release();
     }
-    /** Returns whether the resource exists. */
-    @property bool isEmpty() const
-    {
-        return _p is null;
-    }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
-    Unique release()
+
+    /** Frees the underlying $(D RefT) and nulls it */
+    void release()
     {
         debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+
+            destroy(_p);
+            GC.free(cast(void*)_p);
```

Comment:
If that's also true for RefCounted we should disable such types for now and find a proper solution.
Note that it possible though hacky to set the context pointer, the problem is more how to get the right context ptr. For nested struct you should be able to first initialize a normal value (compiler will fill the context ptr), then move that value. For nested classes you could have a separate argument.